### PR TITLE
Refactor e2e yaml value files

### PR DESCRIPTION
**Pull request by howardjohn**
This change involes:
* renaming uses of old make target
* adding all generated files to gitignore
* create new target to build all e2e yaml files and another for the demo
files that are included in release
* move all testing value files, and example value files, to folders
* create value files for tests that were using --set

See https://github.com/istio/istio/issues/11961 for motivation

### CODE REVIEW

**Comment by howardjohn** [comment]
@mandarjog @sdake

File: Makefile [hunk 1]
```diff
@@ -668,50 +658,26 @@ isti%.yaml: $(HELM) $(HOME)/.helm
 		--values install/kubernetes/helm/istio/values-$@ \
```
Comment (mandarjog) [inline Makefile]: were you suggesting in your proposal to have these values in releases directory ?

File: Makefile [hunk 2]
```diff
@@ -668,50 +658,26 @@ isti%.yaml: $(HELM) $(HOME)/.helm
...
+			istio-one-namespace-trust-domain.yaml \
+			istio-minimal.yaml \
+			istio-multicluster.yaml \
+
+.PHONY: generate_e2e_yaml
+generate_e2e_yaml: $(e2e_files)
+
+# Create yaml files for e2e tests. Applies values-e2e.yaml, then values-$filename.yaml
+$(e2e_files): $(HELM) $(HOME)/.helm istio-init.yaml
+	echo "todo"
```
Comment (mandarjog) [inline Makefile]: Is there anything left to be done here?

**Comment by rshriram** [comment]
This cleanup is too late in the release cycle isn't it? I dont mind, but you have to have a concurrent PR to update all the docs in istio.io because they all link to these yamls directly. I dont want to merge this without seeing a docs PR that fixes all the references (things like minimal istio install, multicluster install, etc. will be broken).

**Comment by howardjohn** [comment]
@rshriram the docs surprisingly didn't need much updates. PR here to update them: https://github.com/istio/istio.io/pull/3412

**Comment by mandarjog** [comment]
John, there are more files you have missed. check location of minimal-values, it also needs to be in the release.

```
docs/examples/multicluster/gke/index.md:150:      --set global.remoteTelemetryAddress=${TELEMETRY_POD_IP} > $HOME/istio-remote.yaml
docs/examples/multicluster/gke/index.md:161:$ kubectl apply -f $HOME/istio-remote.yaml
docs/examples/multicluster/split-horizon-eds/index.md:149:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:108:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:111:    --set global.remoteTelemetryAddress=${TELEMETRY_POD_IP} > $HOME/istio-remote.yaml
docs/setup/kubernetes/multicluster-install/vpn/index.md:132:    $ kubectl apply -f $HOME/istio-remote.yaml
docs/setup/kubernetes/multicluster-install/vpn/index.md:170:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:325:$ kubectl delete -f $HOME/istio-remote.yaml
docs/setup/kubernetes/multicluster-install/vpn/index.md:364:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:594:      --values install/kubernetes/helm/istio/values-istio-remote.yaml \
✘-INT ~/GOHOME/src/istio.io/istio.github.io/content [master|✔]
07:56 $ gr values-
about/notes/1.0.1/index.md:25:- Introduced the convenient `values-istio-gateway.yaml` configuration that enables users to run standalone gateways.
docs/examples/multicluster/split-horizon-eds/index.md:149:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/minimal-install/index.md:37:      --values install/kubernetes/helm/istio/values-istio-minimal.yaml >> $HOME/istio-minimal.yaml
docs/setup/kubernetes/minimal-install/index.md:81:      --values install/kubernetes/helm/istio/values-istio-minimal.yaml
docs/setup/kubernetes/multicluster-install/gateways/index.md:63:        -f @install/kubernetes/helm/istio/values-istio-multicluster-gateways.yaml@ > $HOME/istio.yaml
docs/setup/kubernetes/multicluster-install/vpn/index.md:108:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:170:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:364:    --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/multicluster-install/vpn/index.md:594:      --values install/kubernetes/helm/istio/values-istio-remote.yaml \
docs/setup/kubernetes/quick-start-ibm/index.md:67:    The Istio `demo` profile (`install/kubernetes/helm/istio/values-istio-demo.yaml`) is specified in the following command to support the IBM Cloud Kubernetes Service free cluster, which only contains a single worker providing fewer resources than needed.
docs/setup/kubernetes/quick-start-ibm/index.md:72:    $ helm install install/kubernetes/helm/istio --name istio --namespace istio-system --values install/kubernetes/helm/istio/values-istio-demo.yaml
docs/tasks/security/auth-sds/index.md:58:    $ helm template install/kubernetes/helm/istio --name istio --namespace istio-system --values install/kubernetes/helm/istio/values-istio-sds-auth.yaml >> istio-auth-sds.yaml
docs/tasks/security/vault-ca/index.md:30:        --values install/kubernetes/helm/istio/values-istio-example-sds-vault.yaml \
docs/tasks/security/vault-ca/index.md:41:The yaml file [`values-istio-example-sds-vault.yaml`]({{< github_file >}}/install/kubernetes/helm/istio/values-istio-example-sds-vault.yaml)
```

**Comment by howardjohn** [comment]
> John, there are more files you have missed. check location of minimal-values, it also needs to be in the release.

Just noticed that too when I was making the docs PR. The last commit makes sure istio-minimal is included in release. And looking at that grep (and mine) I think that should be everything.

**Comment by mandarjog** [comment]
My grep exlcuder Chinese content, please check that as well.

@geeknoid is it possible to set an alternate branch for the istio site in a docs pr? Several files are moving around and it would be good for the docs urlnchecker to use this PR branch to check the related docs pr against

**Comment by howardjohn** [comment]
> My grep exlcuder Chinese content, please check that as well.

I got those, thanks

**Comment by mandarjog** [comment]
@rshriram this is ready to go, and there is an accompanying doc PR.

**Comment by howardjohn** [comment]
@rshriram @mandarjog has changes built on this that he wants to get in. Are you ok with merging this in? We have verified that the docs have been fully updated for this change (https://github.com/istio/istio.io/pull/3412) and that the correct files are present when building a release.

**Pull request merged**